### ISSUE TITLE
Update SOA serial with each change

### ISSUE THREAD

**Issue by quartzmo** [GoogleCloudPlatform/gcloud-ruby#320]
The SOA record serial number must be incremented for every recordset change. The gcloud command-line tool automates this as shown in the generated transaction below, in which `0` is incremented to `1`.

```sh
$ gcloud dns record-sets transaction start --zone example-zone
Transaction started [transaction.yaml].
$ cat transaction.yaml 
---
additions:
- kind: dns#resourceRecordSet
  name: example.net.
  rrdatas:
  - ns-cloud-b1.googledomains.com. dns-admin.google.com. 1 21600 3600 1209600 300
  ttl: 21600
  type: SOA
deletions:
- kind: dns#resourceRecordSet
  name: example.net.
  rrdatas:
  - ns-cloud-b1.googledomains.com. dns-admin.google.com. 0 21600 3600 1209600 300
  ttl: 21600
  type: SOA
```

Cloud DNS uses a simple integer serial number, but it is possible to use a date format, as described in [this article](http://www.zytrax.com/books/dns/ch8/soa.html). The complete story of serial numbers is in [RFC 1982](https://tools.ietf.org/html/rfc1982). I propose only supporting the simple integer values that are the default in Cloud DNS.

If the record set change already contains SOA records, I think we should assume the user has taken responsibility for incrementing the serial number, and not attempt it.

**Comment by blowmage** [comment]
@jgeewax Thoughts on this type of functionality? We think this is something that would be really helpful.

**Comment by quartzmo** [comment]
I'm going to go ahead and add this feature today. Its inclusion in the command-line tool is strong evidence of its value.

**Comment by quartzmo** [comment]
Closed by PR #323.
